packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+    });
+
+    it('surfaces [selects="single"] selection on Virtualized Table', async () => {
+        // TODO fix the args
+        const el = await fixture<Table>(
+            virtualizedSingle({
+                ...virtualizedSingle.args,
+                onChange: () => {
+                    return;
+                },
+            } as Properties)
+        );
+
+        expect(el.selected, "'Row 1 selected").to.deep.equal(['0']);
+    });
+
+    it('surfaces [selects="multiple"] selection', async () => {
+        const el = await fixture<Table>(selectsMultiple());
+
+        expect(el.selected, 'Rows 1 and 2 selected').to.deep.equal([
+            'row1',
+            'row2',
+        ]);
+    });
+
+    it('surfaces [selects="multiple"] selection on Virtualized Table', async () => {
+        const el = await fixture<Table>(
+            virtualizedMultiple(virtualizedMultiple.args as Properties)
+        );
+
+        expect(el.selected, 'Rows 1 and 2 selected').to.deep.equal(['0', '48']);
+    });
+
+    it('selects a user-passed value for .selected array with no [selects="single"] specified, but does not allow interaction afterwards', async () => {
+        const el = await fixture<Table>(html`
+            <sp-table size="m" .selected=${['row3']}>
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+                <sp-table-body style="height: 120px">
+                    <sp-table-row value="row1" class="row1">
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                    </sp-table-row>
+                    <sp-table-row value="row2" class="row2">
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                    </sp-table-row>
+                    <sp-table-row value="row3" class="row3">
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                    </sp-table-row>
+                </sp-table-body>
+            </sp-table>
+        `);
+        await elementUpdated(el);
+
+        expect(el.selected.length).to.equal(1);
+
+        const rowThree = el.querySelector('.row3') as TableRow;
+        const rowTwo = el.querySelector('.row2') as TableRow;
+        const rowTwoCheckbox = rowTwo.querySelector('sp-table-checkbox-cell');
```

Comment:
Probably good to `querySelectAll` from the Table here, ensures NOTHING has checkboxes.